### PR TITLE
Install and config husky, lint-staged and commitlint

### PR DESCRIPTION
**Pull request by m-davoodi**

### CODE REVIEW

File: .lintstagedrc.json [hunk 1]
```diff
@@ -0,0 +1,3 @@
+{
+  "*.{js,jsx,ts,tsx,json,css,scss,md,yml}": "prettier --write"
```
Comment (mdeljavan) [inline .lintstagedrc.json]:
```suggestion
  "*.{js,jsx,ts,tsx,json,css,scss,md,yml}": "yarn prettier --write"
```

File: .husky/pre-commit [hunk 2]
```diff
@@ -0,0 +1,6 @@
+#!/usr/bin/env sh
+. "$(dirname -- "$0")/_/husky.sh"
+
+yarn lint-staged && \
+yarn nx affected --target=type-check && \
```
Comment (mdeljavan) [inline .husky/pre-commit:5]: IMO define a script in package.josn

File: .husky/pre-commit [hunk 3]
```diff
@@ -0,0 +1,6 @@
...
+yarn nx affected --target=lint
```
Comment (mdeljavan) [inline .husky/pre-commit:6]: also for this

File: .lintstagedrc.json (hunk 1 above)
Comment (m-davoodi) [reply .lintstagedrc.json]: I add this file according to Lintstage's document and it works properly.

Comment (mdeljavan) [reply .lintstagedrc.json]: In this case prettier run from global that may have different version of our project. when we add yarn we urge that use our prettier.

**Review by mdeljavan — APPROVED** [review]

**Pull request merged**